sample/src/main/java/io/instaleap/statusprogressbar/MainViewModel.kt

```diff
@@ -0,0 +1,127 @@
+package io.instaleap.statusprogressbar
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class MainViewModel : ViewModel() {
+
+    private val _dataModelView = MutableLiveData<DataModelView>()
+    val dataModelView: LiveData<DataModelView> = _dataModelView
+
+    private val _dataList = MutableLiveData<List<DataModelView>>()
+    val dataList: LiveData<List<DataModelView>> = _dataList
+
+    private val _pendingQuantity = MutableLiveData(10)
+    private val _addedQuantity = MutableLiveData(0)
+    private val _removedQuantity = MutableLiveData(0)
+
+    private val dataModelStateList: MutableList<DataModelState> = mutableListOf()
+
+    init {
+        updateDataList()
+        setListDataModelView()
+    }
+
+    fun addItem() {
+        _addedQuantity.value = (_addedQuantity.value ?: 0) + 1
```

Comment:
looks like complex to understand and duplicated code, pls move this to another function